test/LLVMIntrinsics/memcpy_opaque.ll

```diff
@@ -0,0 +1,95 @@
+; RUN: clspv-opt %s -o %t.ll --passes=replace-llvm-intrinsics -opaque-pointers
+; RUN: FileCheck %s < %t.ll
+
+target datalayout = "e-p:32:32-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024"
+target triple = "spir-unknown-unknown"
+
+%struct.outer = type { [2 x %struct.inner] }
+%struct.inner = type { <4 x i32>, <4 x i64> }
+
+define dso_local spir_kernel void @fct1(ptr addrspace(1) nocapture writeonly align 16 %dst, ptr addrspace(1) nocapture readonly align 16 %src) {
+entry:
+  tail call void @llvm.memcpy.p1.p1.i32(ptr addrspace(1) noundef align 16 dereferenceable(16) %dst, ptr addrspace(1) noundef align 16 dereferenceable(16) %src, i32 128, i1 false)
+  ret void
+}
+
+define dso_local spir_kernel void @fct2(ptr addrspace(1) nocapture writeonly align 16 %dst, ptr addrspace(1) nocapture readonly align 16 %src) {
+entry:
+  tail call void @llvm.memcpy.p1.p1.i32(ptr addrspace(1) noundef align 16 dereferenceable(16) %dst, ptr addrspace(1) noundef align 16 dereferenceable(16) %src, i32 16, i1 false)
+  ret void
+}
+
+define dso_local spir_kernel void @fct3(ptr addrspace(1) nocapture writeonly align 16 %dst, ptr addrspace(1) nocapture readonly align 16 %src) {
+entry:
+  tail call void @llvm.memcpy.p1.p1.i32(ptr addrspace(1) noundef align 16 dereferenceable(16) %dst, ptr addrspace(1) noundef align 16 dereferenceable(16) %src, i32 4, i1 false)
+  ret void
+}
+
+define dso_local spir_kernel void @fct4(ptr addrspace(1) nocapture writeonly align 16 %dst, ptr addrspace(1) nocapture readonly align 16 %src) {
+entry:
+  tail call void @llvm.memcpy.p1.p1.i32(ptr addrspace(1) noundef align 16 dereferenceable(16) %dst, ptr addrspace(1) noundef align 16 dereferenceable(16) %src, i32 3, i1 false)
+  ret void
+}
+
+define dso_local spir_kernel void @fct5(ptr addrspace(1) nocapture writeonly align 16 %dst, ptr addrspace(1) nocapture readonly align 16 %src) {
+entry:
+  tail call void @llvm.memcpy.p1.p1.i32(ptr addrspace(1) noundef align 16 dereferenceable(16) %dst, ptr addrspace(1) noundef align 16 dereferenceable(16) %src, i32 64, i1 false)
+  %0 = getelementptr inbounds %struct.outer, ptr addrspace(1) %dst, i32 0, i32 0, i32 1, i32 0
+  %1 = getelementptr inbounds %struct.outer, ptr addrspace(1) %src, i32 0, i32 0, i32 1, i32 0
+  ret void
+}
```

Comment:
What a version of this test like:
```
%0 = getelementptr %struct.outer, ptr addrspace(1) %src, i32 0, i32 0, i32 %n, i32 1
%1 = getelementptr %struct.inner, ptr addrspace(1) %dst, i32 1
call void @llvm.memcpy.p1.p1.i32(ptr addrspace(1) %1, ptr addrspace(1) %0, i32 16, i1 false)
```

Reply:
I'm not sure whether you wanted the dst gep to be `i32 0, i32 1`. Or if you want to have a memcpy with input of different types. I have added a test with a `i32 0, i32 1` in the dst gep.